src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -687,26 +706,59 @@ private Mqtt5Client createCrtClient() throws MessageClientException {
     }
 
     void setClient(Mqtt5Client client) {
-        if (isSSL()) {
-            mqttClientKeyStore.listenToCAUpdates(onKeyStoreUpdate);
-        }
         this.client = client;
     }
 
     private boolean isSSL() {
         return "ssl".equalsIgnoreCase(brokerUri.getScheme());
     }
 
-    void reset() { // TODO callback shouldn't be synchronous
-        stop();
-        try {
-            setClient(createCrtClient());
-        } catch (MessageClientException e) {
-            // TODO recover
-            LOGGER.atError().cause(e).log("unable to start mqtt client during reset");
-            return;
+    /**
+     * Stop and start the mqtt client.
+     */
+    public void reset() {
+        synchronized (resetLock) {
+            closeClient();
```

Comment:
So what's going on with publish/subscribe requests while we're doing the reset?

Reply:
subscribes will get retried after connection success, but publishes will be dropped

Reply:
bridge doesn't spooler LocalMqtt by itself, im not sure if we had that with paho before

Reply:
not just straight up dropped though, I'd expect some gross exceptions to happen. Let's try and make that cleaner by checking if the client exists (nonnull) and is connected before attempting to publish. And log a nicer debug message to say why.

Add an info log in reset to say that messages will be dropped until the connection is back up so that we're clear to customers and ourselves about what is going to happen.

Reply:
we already check for connection before publish, but no logs, will add

Reply:
OK, that's fine. I said debug, but I think trace for that since it is on every single message.